identity.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/bitmark-inc/autonomy-pod-controller/key"
+	"github.com/bitmark-inc/autonomy-pod-controller/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const PrivateKeyLen = 32
+
+// KeyFile is an object to save private
+type KeyFile struct {
+	PrivateKey string `json:"private_key"`
+}
+
+// PodIdentity is an identity object
+type PodIdentity struct {
+	PrivateKey []byte
+	DID        string
+}
+
+// Auth do authentication from the autonomy API server as the role autonomy-pod
+// and get a JWT token for sending and receiving messages
+func (p *PodIdentity) Auth() (string, error) {
+	nowString := fmt.Sprint(int64(time.Now().UnixNano()) / int64(time.Millisecond))
+	signature, err := key.Sign(p.PrivateKey, nowString+"autonomy-node")
+	if err != nil {
+		log.WithField("signature", signature).Error("sign error")
+		return "", err
+	}
+
+	req := map[string]string{
+		"signature": signature,
+		"requester": p.DID,
+		"timestamp": nowString,
+		"Role":      "autonomy-node",
```

Comment:
```suggestion
		"role":      "autonomy-node",
```

Reminder: update `policy.csv` for the API server.